bigtable_input.py

```diff
@@ -239,28 +269,44 @@ def parse(r):
         return sorted([parse(r) for r in rows], key=operator.itemgetter(0))
 
     def delete_row_range(self, format_str, start_game, end_game):
+        """Delete rows related to the given game range.
+
+        Args:
+          format_str:  a string to `.format()` by the game numbers
+            in order to create the row prefixes.
+          start_game:  the starting game number of the deletion.
+          end_game:  the ending game number of the deletion.
+        """
         row_keys = make_single_array(
             self.tf_table.keys_by_range_dataset(
                 format_str.format(start_game),
                 format_str.format(end_game)))
         row_keys = list(row_keys)
+        if not row_keys:
+            utils.dbg('No rows left for games %d..%d' % (
+                start_game, end_game))
+            return
         utils.dbg('Deleting %d rows:  %s..%s' % (
             len(row_keys), row_keys[0], row_keys[-1]))
-        rows = [self.bt_table.row(k) for k in row_keys]
+        batches = []
         batch = []
-        def mutate_batch():
-            nonlocal batch
-            utils.dbg('...deleting batch %s..%s' % (
-                batch[0].row_key, batch[-1].row_key))
-            self.bt_table.mutate_rows(batch)
+        total_keys = len(row_keys)
+        table_name = self.bt_table.name
+        while row_keys:
```

Comment:
This block of code is identical to utils.iter_chunks

https://github.com/tensorflow/minigo/blob/master/utils.py#L59

replace with `batches = list(utils.iter_chunks(bigtable.row.MAX_MUTATIONS, row_keys))

Reply:
The comments mention that the reverse() is important-ish, but I'm not sure how that works - sholudn't the multiprocessing make order nondeterministic?

Reply:
Thanks for the tip! I'll swap that in.

The deleting in reverse is to allow for a more sensible end state if you change your mind (say, due to a mistake in the timestamp) and abort the process: you'll have a little trimmed from the end, rather than a little trimmed out of the middle. I'll add a comment to that effect.

I can reverse the list of keys before calling `utils.iter_chunks` and preserve that behavior.